board/tests.py

```diff
@@ -146,4 +149,178 @@ def test_post_creation_combinations(self):
+            )
+            p.save()
+            posts.append(p)
+        return posts
+
+    @tag('core')
+    def test_get_posts(self):
+        """Make sure that it returns proper JSON responses with proper amount of posts."""
+        
+        b = Board(title='hi', description='hello', bg=Image(name='i', photo=bytearray('hi', 'utf-8')).save())
+        b.save()
+        posts = self.add_posts(b, 30)
+        
+        # First 10 posts
+        res1 = self.client.get(
+            reverse('board:posts-get'),
+            {
+                'board': str(b.uuid),
+                'index': '0',
+                'amount': '10',
+            },
+            HTTP_ACCEPT='application/json',
+        )
+        exp1 = []
+        for i in range(10):
+            p = posts[-i-1]
+            exp1.append(GetPosts.get_post_dict(p))
+        self.assertEqual(res1.json(), exp1)
+
+
+        # Second 10 posts
+        res2 = self.client.get(
+            reverse('board:posts-get'),
+            {
+                'board': str(b.uuid),
+                'index': '10',
+                'amount': '10',
+            },
+            HTTP_ACCEPT='application/json',
+        )
+        exp2 = []
+        for i in range(10):
+            p = posts[-i-11]
+            exp2.append(GetPosts.get_post_dict(p))
+        
+        self.assertEqual(res2.json(), exp2)
+
+    @tag('core')
+    def test_get_posts_fully_loaded(self):
+        """
+        Test for when the client requests more posts than what the database
+        have available. It should return back just the sufficient amount.
+        """
+
+        b1 = Board(title='hi', description='hello', bg=Image(name='i', photo=bytearray('hi', 'utf-8')).save())
+        b2 = Board(title='hi2', description='hello2', bg=Image(name='i2', photo=bytearray('hi', 'utf-8')).save())
+
+        b1.save()
+        b2.save()
+        posts1 = self.add_posts(b1, 5)
+
+        # board 1 has only 5 posts but the client requested 10
+        res1 = self.client.get(
+            reverse('board:posts-get'),
+            {
+                'board': str(b1.uuid),
+                'index': '0',
+                'amount': '10',
+            },
+            HTTP_ACCEPT='application/json',
+        )
+        exp1 = []
+        for i in range(5):
+            exp1.append(GetPosts.get_post_dict(posts1[-i-1]))
+        self.assertEqual(res1.json(), exp1)
+
+        # board 2 has no posts but the client requested 10
+        res2 = self.client.get(
+            reverse('board:posts-get'),
+            {
+                'board': str(b2.uuid),
+                'index': '5',
+                'amount': '10',
+            },
+            HTTP_ACCEPT='application/json',
+        )
+        exp2 = []
+        self.assertEqual(res2.json(), exp2)
+
+
+    def test_get_posts_invalid_query(self):
+        """
+        Test for missing or invalid query string such as:
+        - negative numbers 
+        - non-existent board uuid
+        - etc
```

Comment:
This should be rewritten as one summary statement, and if not, it should be written like:

```
"""Test for things.

This test will test for:
- c
- b
- c
```

Reply:
new issue 😄